samples/weather-kotlin/src/main/kotlin/Weather.kt

```diff
@@ -0,0 +1,118 @@
+package org.acme
+
+import io.quarkiverse.mcp.server.Tool
+import io.quarkiverse.mcp.server.ToolArg
+import io.quarkus.qute.Qute
+import io.quarkus.rest.client.reactive.Url
+import jakarta.ws.rs.GET
+import jakarta.ws.rs.Path
+import org.eclipse.microprofile.rest.client.inject.RegisterRestClient
+import org.eclipse.microprofile.rest.client.inject.RestClient
+import org.jboss.resteasy.reactive.RestPath
+
+class Weather {
+    @RestClient
+    lateinit var weatherClient: WeatherClient
```

Comment:
Constructor injection and final fields would probably be better
```suggestion
class Weather(
    @RestClient
    private val weatherClient: WeatherClient
) {
```